solr/core/src/test/org/apache/solr/request/TestRemoteStreaming.java

```diff
@@ -116,8 +116,10 @@ public void testNoUrlAccess() throws Exception {
     SolrQuery query = new SolrQuery();
     query.setQuery("*:*"); // for anything
     query.add("stream.url", makeDeleteAllUrl());
-    SolrException se = expectThrows(SolrException.class, () -> getSolrClient().query(query));
-    assertSame(ErrorCode.BAD_REQUEST, ErrorCode.getErrorCode(se.code()));
+    try (SolrClient solrClient = getHttpSolrClient(getServerUrl())) {
```

Comment:
so, there was a discussion that we should always use the try/with resources pattern with SolrClient...   So I moved in the direction, with maybe the idea that the SolrJettyTestBase.getSolrClient() might go away, because we would always create our own and then close it, instead of having the base test class provide one and be responsible for it...   

There is the createNewSolrClient method, so I am swapping to that.    

We do need a better policy, ideally written down and/or code checked on if we use base classes to manage lifecycle of SolrClient in tests or do the try/with resources...

Reply:
If you have a *new* SolrClient, then yes -- we should use try-with-resources.  Ideally any utility methods that create a new one would have "create" or "new" in their name; alas this isn't true.  If you have some SolrClient managed by test infrastructure somehow, then there we shouldn't close it.  I don't think we should mandate either path.